h2/src/main/org/h2/mode/PgCatalogTable.java

```diff
@@ -544,19 +546,34 @@ public PgCatalogTable(Schema schema, int id, int type) {
                 if (pgType == PgServer.PG_TYPE_UNKNOWN || !types.add(pgType)) {
                     continue;
                 }
+                String name;
+                int elemId;
+                switch (pgType) {
+                case PgServer.PG_TYPE_TEXTARRAY:
```

Comment:
Please, merge changes from the `master` branch and use `PG_TYPE_VARCHAR_ARRAY` / 1015 here.

You can add support for more types of array, but it requires some additional work.